setup.cfg

```diff
@@ -106,7 +107,7 @@ install_requires =
     graphviz>=0.12
     gunicorn>=19.5.0
     httpx
-    importlib_metadata~=1.7;python_version<"3.9" # We could work with 3.1, but argparse needs <2
+    importlib_metadata~=3.1;python_version<"3.9"
```

Comment:
@ashb - I guess the argparse limitation is gone already ? I am not quite sure what "argparse" probiem it was (the only argparse which is not-built in is the sphinx-argparse).

Reply:
Hmmmm, I'm afraid I can't remember, but sphinx-argparse doesn't use importlib etc.

If it works and passes tests :+1:

Reply:
I think you meant argcomplete; it has a importlib-metadata pin (and pip would get this wrong before the resolver upgrade). argcomplete was updated to be compatible with `importlib-metadata<5` now though, so this should work. (Maybe we should look into importlib-metadata 4.x compatibility as well.)

Reply:
I actually changed it to >1.7 now to handle current constraint files, so we should automatically get latest compatible version.

BTW. Should not we install `importlib-metadata` regardless of python version ? This is what jira package does anyway and it seems it gets updated more frequently than Python 3.9 is. We might - at some point in time start relying on some features there. Seems that importlib_metadata 4.4 for example is only available in 3.10 (https://github.com/python/importlib_metadata). 

Maybe we should just remove `;python<3.9`

Reply:
It also makes less 'moving parts' - with < 3.9 we need to handle (and actually should test) 3 cases:
* < 3.9 + importlib_metadata
* >= 3.9
* >= 3.9  +importlib_metadata

By removing <3.9 we have less variability.

Reply:
The marker was introduced in #12862. I don’t have an opinion either way (both have their advantages and supported by different people).

Reply:
Oh yes, argcomplete sounds right

Reply:
Mostly following the principle that "we don't need any of the modern features here, and less things installed = less to worry about"